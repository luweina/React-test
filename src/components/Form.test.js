import React from "react"
import renderer from 'react-test-renderer'
import Form from './Form'
import * as ReactDOM from "react-dom"
import {render, fireEvent} from "@testing-library/react"
// import { shallow, mount } from 'enzyme';

describe("Form component render properly", ()=> {

    it("render Form correctly", () => {
        const {queryByTestId, queryByPlaceholderText} = render (<Form/>)
        expect(queryByTestId("hi")).toBeTruthy()
        expect(queryByPlaceholderText("Enter somethings")).toBeTruthy()
    })

    it("testing functionalites of the Todo comonpnent", () => {
       const root = document.createElement('div')
       ReactDOM.render(<Form />, root)

       expect(root.querySelector("label").textContent).toBe("What needs to be done?")

    })

})

describe ("snapshot test for Form component", () =>{
    it("snapshot", () => {
        const tree = renderer.create(<Form />).toJSON( );
        expect(tree).toMatchSnapshot()
    })
} )

describe ("input and button functionality test", ( ) => {
    it("update input value ", () => {
        const {queryByPlaceholderText} = render (<Form />)
        const userInput = queryByPlaceholderText ("Enter somethings")
        fireEvent.change(userInput, {target: {value: "test"}})
        expect (userInput.value).toBe("test")
    })

    it(" button dont submit ", () =>{
        const submitButton = jest.fn()
        const {queryByTestId} = render (<Form submitButton = {submitButton} />)
        fireEvent.click(queryByTestId('hi'))
        expect(submitButton).not.toHaveBeenCalled()
    })

})


