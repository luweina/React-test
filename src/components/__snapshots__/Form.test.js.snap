// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test for Form component snapshot 1`] = `
<form
  onSubmit={[Function]}
>
  <h2
    className="label-wrapper"
  >
    <label
      className="label__lg"
      htmlFor="new-todo-input"
    >
      What needs to be done?
    </label>
  </h2>
  <input
    autoComplete="off"
    className="input input__lg"
    id="new-todo-input"
    name="text"
    onChange={[Function]}
    placeholder="Enter somethings"
    type="text"
    value=""
  />
  <button
    className="btn btn__primary btn__lg"
    data-testid="hi"
    type="submit"
  >
    Add
  </button>
</form>
`;
